lib/ui/window.dart

```diff
@@ -145,12 +145,64 @@ class Locale {
+  /// primary language subtag should be two to three lowercase letters. The
+  /// region region subtag should be two uppercase letters or three digits. See
+  /// the [Unicode Language
+  /// Identifier](https://www.unicode.org/reports/tr35/#Unicode_language_identifier)
+  /// specification.
+  ///
+  /// Validity is not checked by default, but some methods may throw away
+  /// invalid data.
+  ///
+  /// See also: [Locale.create].
+  const Locale(
+    this._languageCode, [
+    this._countryCode,
+  ]) : assert(_languageCode != null),
+       this.scriptCode = null,
+       this._variants = null;
+
+  /// Creates a new Locale object.
+  ///
+  /// The keyword arguments specify the subtags of the Locale.
+  ///
+  /// The subtag values are _case sensitive_ and should be valid subtags
+  /// according to CLDR supplemental data:
+  /// [language](http://unicode.org/cldr/latest/common/validity/language.xml),
+  /// [script](http://unicode.org/cldr/latest/common/validity/script.xml),
+  /// [region](http://unicode.org/cldr/latest/common/validity/region.xml) and
+  /// [variant](http://unicode.org/cldr/latest/common/validity/variant.xml). If
+  /// there is more than one variant, they should be in sorted order. This list
+  /// will be used as-is, and should never again be modified.
+  ///
+  /// Well-formedness is not fully checked by this constructor: language, script
+  /// and region subtags with bad length are discarded, however characters are
+  /// not checked. Other methods may throw away subtags with syntactically
+  /// invalid contents.
+  const Locale.fromSubtags({
+    String language = 'und',
+    String script,
+    String region,
+    List<String> variants,
+  }) : _languageCode = (language == null || language.length == 4 ||
+                        language.length < 2 || language.length > 8)
+                       ? 'und' : language,
+       // Essentially we'd like to catch errors in debug mode:
+       // assert(_languageCode == language):
+       assert(language == null ||
+              (language.length != 4 && language.length >= 2 && language.length <= 8)),
+       scriptCode = (script == null || script.length != 4) ? null : script,
+       assert(script == null || script.length == 4),
+       _countryCode = (region == null || region.length < 2 || region.length > 3) ? null : region,
+       assert(region == null || (region.length >= 2 && region.length <= 3)),
+       // We can check "variants[0].length >= 4" in a const expression, but
+       // there's not much we can do in response except drop all variants.
+       _variants = (variants == null || variants.length == 0) ? null : variants, // ignore: prefer_is_empty, const_eval_type_bool_num_string
+       assert(variants == null || variants.length > 0); // ignore: prefer_is_empty, const_eval_type_bool_num_string, https://github.com/dart-lang/sdk/issues/34798, https://dart-review.googlesource.com/c/sdk/+/71340
```

Comment:
we definitely should not do this kind of thing. The release mode code should have no branches and be super fast.
If you want to validate the language code you should do it before calling the constructor.

Reply:
(or after you have the object)